R/sw_Vegetation.R

```diff
@@ -928,286 +620,661 @@ get_season_description_v2 <- function(x) {
+  psps <- list(
+    t_ref = splines::periodicSpline(
+      ref_temp_norm ~ mon_norm_center,
+      period = yr_std_d
+    ),
 
-    # Identify months of sub-period `p` in reference
-    ids <- circ_seq(
-      from = ref_desc[iprev],
-      to = ref_desc[p],
-      int = 12
+    t_target = splines::periodicSpline(
+      target_temp_norm ~ mon_norm_center,
+      period = yr_std_d
+    ),
+
+    veg = splines::periodicSpline(
+      veg_norm ~ mon_norm_center,
+      period = yr_std_d
+    ),
+
+    veg_asif = splines::periodicSpline(
+      veg_asif_norm ~ mon_norm_center,
+      period = yr_std_d
     )
+  )
+
+  itcurves <- 1:2
+  ivcurves <- 3:4
+
+
+  #------ Step 3) Correct vegetation
+  # based on radial temperature expansion/contraction
+
+  # Prepare container for adjusted (normalized) vegetation points (xy)
+  vadj <- vector("list", length = Nadj)
+  k_vadj <- 1
+  ksc <- 1 # counter of continous adjustment sections
+  kce <- c(1, 0, 0) # counter of continuous failure sections
 
-    n_ids <- length(ids)
+  # Loop through points along season-ellipses (`ptadj`)
+  for (k in seq_len(Nadj)) {
 
-    # Target sub-period duration
-    n_ids2 <- n_ids + desc_diffs[p] - desc_diffs[iprev]
+    #--- Locate intersection between target temperature function and
+    # line connecting points (time of year, reference temperature function) and
+    # season center points
+    if (abs(ptadj[k, 3] - ptadj[k, 1]) > rSW2_glovars[["tol"]]) {
+
+      # Function describing line that is connecting the circle-center point
+      # with the k-th point on the season-circle
+      fline_radial <- rSW2utils::f_2pline(
+        x0 = ptadj[k, 1], y0 = ptadj[k, 2],
+        x1 = ptadj[k, 3], y1 = ptadj[k, 4]
+      )
+
+      # Search limits for locating intersections
+      if (ptadj[k, 3] > ptadj[k, 1]) {
+        # limits: ellipse center to max influence
+        olims <- c(ptadj[k, 1], ptadj[k, 6])
+      } else {
+        # limits: min influence to ellipse center
+        olims <- c(ptadj[k, 5], ptadj[k, 1])
+      }
+      olims <- rSW2utils::extend_range(olims, 1 + ptadj[k, 7] / 2)
+
+      # Locate the x-axis value where line crosses
+      #   (i) normalized reference temperature curve
+      #   (ii) normalized target temperature curve
+      xcrosses_tcurves <- lapply(
+        X = itcurves,
+        FUN = function(itp) {
+          rSW2utils::uniroots(
+            f = function(x) {
+              predict(psps[[itp]], x)[["y"]] - fline_radial(x)[, 2]
+            },
+            xlim = olims,
+            tol = tol,
+            expected_nroots = 1
+          )
+        }
+      )
+
+      # Success of detection of temperature curves intersections
+      good_tcurves <- sapply(
+        xcrosses_tcurves,
+        rSW2utils::has_uniroots,
+        tol = tol2
+      )
+
+      if (all(good_tcurves)) {
+        # Locate (all) x-axis values where line crosses
+        #   (iii) nomarlized vegetation curves
```

Comment:
normalized

Reply:
Thanks for catching the typo! Fixed with da89ddd04f73f095698765cc54ad17af8b16c961